src/providers/NotificationProvider.tsx

```diff
@@ -0,0 +1,58 @@
+import { createContext, ReactNode, useContext, useState } from 'react';
+import {
+  NotificationAlert,
+  NotificationAlertProps,
+} from '../components/NotificationAlert.tsx';
+
+interface NotificationContextData {
+  setNotification: (notification: NotificationAlertProps | null) => void;
+}
+
+const NotificationContext = createContext<NotificationContextData>({
+  setNotification: () => {},
+});
+
+export const useNotificationContext = () => {
+  const context = useContext(NotificationContext);
+  if (!context) {
+    throw new Error(
+      'useNotificationContext must be used within a NotificationProvider',
+    );
+  }
+  return context;
+};
+
+function NotificationProvider({ children }: { children: ReactNode }) {
+  const [notification, setNotification] =
+    useState<NotificationAlertProps | null>(null);
+
+  const handleSetNotification = (
+    notification: NotificationAlertProps | null,
+  ) => {
+    setNotification(notification);
+
+    if (notification !== null) {
+      setTimeout(() => {
+        setNotification(null);
+      }, 5000);
+    }
+  };
+
+  return (
```

Comment:
This is fine from upskill perspective, but it can be pretty buggy in real life :D let's say you call setNotification twice, and the second time was after 4s. It will update the value but will close after 1s :D 
We can discuss on the final meeting how to improve this ;)